### PR TITLE
Ability to specify `arg_options` in the schema to override the arguments

### PR DESCRIPTION
**Pull request by joehoyle**

### CODE REVIEW

File: tests/test-rest-controller.php [hunk 1]
```diff
@@ -73,4 +73,12 @@ function test_validate_schema_format_date_time() {
 		);
 	}
 
+	function test_get_endpoint_args_for_item_schema_arg_options() {
+
+		$controller = new WP_REST_Test_Controller();
+		$args = $controller->get_endpoint_args_for_item_schema();
+
+		$this->assertEquals( false, $args['someargoptions']['required'] );
```
Comment (rachelbaker) [inline tests/test-rest-controller.php]: @joehoyle can this just use `assertFalse()`?

Comment (joehoyle) [inline tests/test-rest-controller.php]: I could - given that I was comparing the equality in a few fields - it kind of meant sense to keep it consistent with that.

Comment (joehoyle) [inline tests/test-rest-controller.php]: Fixed in 5ac10a6

**Comment by rachelbaker** [comment]
Merged #1345

**Pull request merged**